src/figuresClasses.ts

```diff
@@ -1,11 +1,101 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle' = 'triangle';
+
+  color: 'red' | 'green' | 'blue';
+
+  private sideA: number;
+
+  private sideB: number;
+
+  private sideC: number;
+
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Sides of triangle should be greater than 0');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('One of three sides is bigger than sum of two others');
+    }
+
+    this.color = color;
+    this.sideA = a;
+    this.sideB = b;
+    this.sideC = c;
```

Comment:
You don't need to manually assign the constructor parameters to class properties when using TypeScript. You can declare the properties directly in the constructor signature with access modifiers to simplify the code.